main/autogen/constant_hash.h

```diff
@@ -0,0 +1,34 @@
+#ifndef AUTOGEN_CONSTANT_HASH_H
+#define AUTOGEN_CONSTANT_HASH_H
+
+#include <string>
+#include <vector>
+
+#include "core/GlobalState.h"
+
+namespace sorbet::parser {
+// pre-declaring this instead of importing this because including
+// "parser/Node.h" more than once causes errors
```

Comment:
I fixed the include issues in #5576, I think, but I also think predeclaring `Node` is better than pulling in a bunch of stuff from `sorbet::parser` in this header.